src/lint_errors.h

```diff
@@ -26,79 +28,107 @@ namespace zetasql {
 
 namespace linter {
 
-enum ErrorCode : int {
-    kLineLimit = 1,
-    kParseFailed = 2,
-    kSemicolon = 3,
-    kUppercase = 4,
-    kCommentStyle = 6,
-    kAlias = 7
+enum class ErrorCode : int {
+  kLineLimit = 1,
+  kParseFailed = 2,
+  kSemicolon = 3,
+  kUppercase = 4,
+  kCommentStyle = 6,
+  kAlias = 7,
+  kUniformTab = 16,
+  kNoIndentTab = 17
 };
 
-// Its a class for saving properties of a one lint error.
+// Stores properties of a single lint error.
 class LintError {
  public:
-    LintError(ErrorCode type, absl::string_view filename,
-        int line )
-        : type_(type), filename_(filename),
-        line_(line) {}
+  LintError(const LintError &result);
 
-    LintError(ErrorCode type, int line)
-        : type_(type), filename_(""), line_(line) {}
+  LintError(ErrorCode type, absl::string_view filename, int line, int column,
+            absl::string_view message)
+      : type_(type),
+        filename_(filename),
+        line_(line),
+        column_(column),
+        message_(message) {}
 
-    absl::string_view GetErrorMessage();
+  absl::string_view GetErrorMessage();
```

Comment:
add documentation for public functions (you can skip constructors, since they are trivial).

Reply:
I was also skipping getter/setters since they were trivial to me. Though, from now on I won't skip these. (Added message for all getter/setters)